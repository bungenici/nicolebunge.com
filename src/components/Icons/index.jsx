import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import instagram from '../../images/instagram.svg';
import linkedin from '../../images/linkedin.svg';
import xing from '../../images/xing.svg';
import styles from './style.module.css';

const Icons = (props) => {
  const { className, ...otherProps } = props;

  return (
    <ul className={classNames(className, styles.icons)} {...otherProps}>
      <li>
        <a
          href="https://www.instagram.com/nicole.bunge/"
          itemProp="sameAs"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img src={instagram} alt="Nicole Bunge auf Instagram" />
        </a>
      </li>
      <li>
        <a
          href="https://www.linkedin.com/in/nicole-bunge-1b891415a/"
          itemProp="sameAs"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img src={linkedin} alt="Nicole Bunge auf LinkedIn" />
        </a>
      </li>
      <li>
        <a
          href="https://www.xing.com/profile/Nicole_Bunge3"
          itemProp="sameAs"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img src={xing} alt="Nicole Bunge auf Xing" />
        </a>
      </li>
    </ul>
  );
};

Icons.propTypes = {
  className: PropTypes.string,
};

Icons.defaultProps = {
  className: '',
};

export default Icons;
