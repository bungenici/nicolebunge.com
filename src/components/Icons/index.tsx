import classNames from 'classnames';
import React, { HTMLAttributes } from 'react';
import instagram from '../../images/instagram.svg';
import linkedin from '../../images/linkedin.svg';
import xing from '../../images/xing.svg';
import styles from './style.module.css';

export type IconsProps = HTMLAttributes<HTMLUListElement>;

const Icons: React.FC<IconsProps> = (props) => {
  const { className, ...otherProps } = props;

  return (
    <ul className={classNames(className, styles.icons)} {...otherProps}>
      <li>
        <a
          href="https://www.instagram.com/nicole.bunge/"
          itemProp="sameAs"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img src={instagram} alt="Nicole Bunge auf Instagram" />
        </a>
      </li>
      <li>
        <a
          href="https://www.linkedin.com/in/nicole-bunge-1b891415a/"
          itemProp="sameAs"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img src={linkedin} alt="Nicole Bunge auf LinkedIn" />
        </a>
      </li>
      <li>
        <a
          href="https://www.xing.com/profile/Nicole_Bunge3"
          itemProp="sameAs"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img src={xing} alt="Nicole Bunge auf Xing" />
        </a>
      </li>
    </ul>
  );
};

export default Icons;
