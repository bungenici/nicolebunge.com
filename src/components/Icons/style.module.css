.icons {
  display: flex;
  list-style: none;
  margin: 0.25em -0.25em 1.25em;
  padding: 0;
}

.icons > li {
  margin: 0.25em;
}
