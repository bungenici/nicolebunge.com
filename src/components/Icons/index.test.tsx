/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Icons from '.';

describe('Icons component', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Icons />);

    expect(asFragment()).toMatchSnapshot();
  });
});
