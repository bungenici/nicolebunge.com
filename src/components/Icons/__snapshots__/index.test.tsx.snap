// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons component renders correctly 1`] = `
<DocumentFragment>
  <ul
    class="icons"
  >
    <li>
      <a
        href="https://www.instagram.com/nicole.bunge/"
        itemprop="sameAs"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Nicole Bunge auf Instagram"
          src="test-file-stub"
        />
      </a>
    </li>
    <li>
      <a
        href="https://www.linkedin.com/in/nicole-bunge-1b891415a/"
        itemprop="sameAs"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Nicole Bunge auf LinkedIn"
          src="test-file-stub"
        />
      </a>
    </li>
    <li>
      <a
        href="https://www.xing.com/profile/Nicole_Bunge3"
        itemprop="sameAs"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Nicole Bunge auf Xing"
          src="test-file-stub"
        />
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
